core/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -389,20 +383,20 @@ Query rangeQuery(String field, Object lowerTerm, Object upperTerm,
         BYTE("byte", NumericType.BYTE) {
             @Override
             Byte parse(Object value, boolean coerce) {
+                double doubleValue = objectToDouble(value);
+
+                if (doubleValue < Byte.MIN_VALUE || doubleValue > Byte.MAX_VALUE) {
+                    throw new IllegalArgumentException("Value [" + value + "] is out of range for a byte");
+                }
+                if (!coerce && doubleValue % 1 != 0) {
```

Comment:
This only rejects `coerce=false` values for numbers with decimals but it won't reject strings, which seems to run contrary to how coercion is supposed to work. I did not change this behaviour. I just wanted to call it out as inconsistent with how the parse methods in AbstractXContentParser work, should this be changed?